app/models/product_user.rb

```diff
@@ -2,9 +2,14 @@ class ProductUser < ActiveRecord::Base
   belongs_to :user
   belongs_to :product
   belongs_to :product_access_group
-  
-  validates_numericality_of :product_id, :user_id, :approved_by, :only_integer => true
-  validates_uniqueness_of :user_id, :scope => :product_id, :message => 'is already approved'
 
-  before_create lambda { |pu| pu.approved_at = Time.zone.now }
-end
\ No newline at end of file
+  validates_numericality_of :product_id, :user_id, :approved_by, only_integer: true
+  validates_uniqueness_of :user_id, scope: :product_id, message: "is already approved"
+
+  before_create -> (product_user) { product_user.approved_at = Time.zone.now }
+
+  def training_request
```

Comment:
It will exist for a very brief time when we create a new `ProductUser`, but for most purposes it's going to be nothing. It's used in `ProductUserCreator.manage_training_request` so it might make more sense to move this code there.